server/server.go

```diff
@@ -3593,3 +3599,42 @@ func (s *Server) updateRemoteSubscription(acc *Account, sub *subscription, delta
 
 	s.updateLeafNodes(acc, sub, delta)
 }
+
+func (s *Server) startRateLimitLogExpiration() {
+	s.mu.Lock()
+	interval := int64(time.Second)
```

Comment:
You set here under lock, but seems not needed. And you never seem to access or use this value, you just overwrite it below when reading from the channel on line 3617.

Reply:
I do see it could be used on line 3622, but could then this just be a variable defined in the Go routine, near where you define the ticker?

Reply:
Again, the interval setting here is not the thing I want the lock for (could move out of the lock). It is the channel.. That being said, I think that the channel should be created in &Server{} when creating NewServer instead.. will revisit that.